Ok, forget that other problem. i forgot to quote '^' in zsh

but i have a package that tries to set CC to the mpi compiler like this:

```
    def configure_args(self):
        spec = self.spec
        extra_args = []

        if '+mpi' in spec:
            extra_args.extend([
                "--enable-mpi",
                "CC=%s" % spec['mpi'].mpicc
                ])
```

When I configure this package, the build log says configure was invoked like so:
```
==> mochi-ssg: Executing phase: 'configure'
==> [2020-12-10-21:52:24.388050] '/tmp/robl/spack-stage/spack-stage-mochi-ssg-0.4.1-stv7lnly3odrryh4mkcxasd33i2iutea/spack-src/configure' '--prefix=/gpfs/mira-home/robl/src/spack/opt/spack/cray-cnl7-mic_knl/gcc-9.3.0/mochi-ssg-0.4.1-stv7lnly3odrryh4mkcxasd33i2iutea' '--enable-mpi' 'CC=/gpfs/mira-home/robl/src/spack/lib/spack/env/gcc/gcc'
```

And that's not an mpi compiler:

```
configure:13005: checking If MPI programs can be compiled
configure:13018: /gpfs/mira-home/robl/src/spack/lib/spack/env/gcc/gcc -o conftest -Wall -Wextra -g -O2   conftest.c  >&5
/opt/cray/pe/cce/10.0.1/binutils/x86_64/x86_64-pc-linux-gnu/bin/ld: /tmp/ccbzkYDI.o: in function `main':
/tmp/robl/spack-stage/spack-stage-mochi-ssg-0.4.1-stv7lnly3odrryh4mkcxasd33i2iutea/spack-src/conftest.c:27: undefined reference to `MPI_Init'
```

--- a/var/spack/repos/builtin/packages/cray-mpich/package.py
+++ b/var/spack/repos/builtin/packages/cray-mpich/package.py
@@ -15,6 +15,11 @@ class CrayMpich(Package):
 
     maintainers = ['haampie']
 
+    version('8.1.0')
+    version('8.0.16')
+    version('8.0.14')
+    version('8.0.11')
+    version('8.0.9')
     version('7.7.16')
     version('7.7.15')
     version('7.7.14')

--- a/var/spack/repos/builtin/packages/cray-mpich/package.py
+++ b/var/spack/repos/builtin/packages/cray-mpich/package.py
@@ -20,7 +20,7 @@ class CrayMpich(Package):
     version('7.7.14')
     version('7.7.13')
 
-    provides('mpi@3:')
+    provides('mpi@3')
 
     def setup_run_environment(self, env):
         env.set('MPICC', spack_cc)

